httpcore5-testing/src/main/java/org/apache/hc/core5/testing/nio/IOReactorExecutor.java

```diff
@@ -85,7 +85,10 @@ public void execute(final IOSession session) {
 
                 }))) {
             if (status.compareAndSet(Status.READY, Status.RUNNING)) {
-                ioReactorRef.get().start();
+                if(this instanceof AsyncServer){
```

Comment:
@xiaohu-zhang This is terrible. Even for test classes we should not be doing things like that.